### ISSUE TITLE
SphinxWarning: autodoc_default_flags is now deprecated, treated as error

### ISSUE THREAD

**Issue by aralmpa** [lsaffre/dblog#1]
Dear @lsaffre ,

I followed your tutorial for making my own blog based on the template you have (dblog). Everything seems to work fine until I execute `inv bd`. 

The command starts to execute and it crashes with the following output:
"""
Invoking Sphinx in directory /home/localaccount/lab_8/blog/docs...
Invoke sphinx-build -b html -T -W . /home/localaccount/lab_8/blog/docs/.build
Running Sphinx v1.8.0
loading translations [en]... done

Traceback (most recent call last):
   File "/home/localaccount/lab_8/venv8/local/lib/python2.7/site-packages/sphinx/cmd/build.py", line 303, in build_main
    args.tags, args.verbosity, args.jobs, args.keep_going)
  File "/home/localaccount/lab_8/venv8/local/lib/python2.7/site-packages/sphinx/application.py", line 251, in __init__
    self.emit('config-inited', self.config)
  File "/home/localaccount/lab_8/venv8/local/lib/python2.7/site-packages/sphinx/application.py", line 510, in emit
    return self.events.emit(event, self, *args)
  File "/home/localaccount/lab_8/venv8/local/lib/python2.7/site-packages/sphinx/events.py", line 80, in emit
    results.append(callback(*args))
  File "/home/localaccount/lab_8/venv8/local/lib/python2.7/site-packages/sphinx/ext/autodoc/__init__.py", line 1565, in merge_autodoc_default_flags
    logger.warning(__('autodoc_default_flags is now deprecated. '
  File "/usr/lib/python2.7/logging/__init__.py", line 1456, in warning
    self.logger.warning(msg, *args, **kwargs)
  File "/usr/lib/python2.7/logging/__init__.py", line 1179, in warning
    self._log(WARNING, msg, args, **kwargs)
  File "/usr/lib/python2.7/logging/__init__.py", line 1286, in _log
    self.handle(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 1296, in handle
    self.callHandlers(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 1336, in callHandlers
    hdlr.handle(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 755, in handle
    rv = self.filter(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 617, in filter
    if not f.filter(record):
  File "/home/localaccount/lab_8/venv8/local/lib/python2.7/site-packages/sphinx/util/logging.py", line 412, in filter
    raise SphinxWarning(message)
SphinxWarning: autodoc_default_flags is now deprecated. Please use autodoc_default_options instead.

Warning, treated as error:
autodoc_default_flags is now deprecated. Please use autodoc_default_options instead.
"""

"localaccount" is an alias for my local user account and not a typo.

My venv has these versions installed: 

- python 2.7.13
- atelier 1.1.8
- Sphinx 1.8.0

Btw, did not spot a "requirements.txt" so I was not sure if had the versions correctly.

I also noticed by some brief research on the error that similar sphinx-related issues were reported (in different context) in the past few days.

Thank you in advance.

**Comment by lsaffre** [comment]
Thanks for reporting. Yes we also had the issue yesterday morning: https://travis-ci.org/lino-framework/book/jobs/430149478
I will now start investigating this.

**Comment by lsaffre** [comment]
In atelier.sphinxconf you can see where I set these default values. Before doing so I test for the Sphinx version:

```
    if LooseVersion(sphinx.__version__) < LooseVersion("1.8"):
        globals_dict.update(autodoc_default_flags=[
            'show-inheritance', 'members'])
    else:
        globals_dict.update(autodoc_default_options={
            'members': None, 'show-inheritance': None})
```
Seems that something doesn't work as expected there...

**Comment by lsaffre** [comment]
@aralmpa, can you try the following code in your Python environment:

```
>>> from distutils.version import LooseVersion
>>> import sphinx
>>> sphinx.__version__
'1.8.0'
>>> LooseVersion(sphinx.__version__)
LooseVersion ('1.8.0')
>>> LooseVersion("1.8.0") < LooseVersion("1.8")
False
```
On my machine these doctests pass.  But on your machine?

**Comment by aralmpa** [comment]
@lsaffre Yes the test passes - The result is False. 

However, I could not find the code you mention above (set default values) anywhere in **atelier.sphinxconf**. 

I expect these lines would be in the __init__ module inside the folder sphinxconf yes?

What is more I cannot find anywhere inside the module `global_dict.update(autodoc_default_options=[...]`. On the contrary, default_**flags** is used.

**Comment by lsaffre** [comment]
The code is here:
https://github.com/lino-framework/atelier/blob/master/atelier/sphinxconf/__init__.py#L106

And oops, here is the explanation: the change that checks for the Sphinx
version was not yet released!  Now I released atelier 1.1.9:
https://pypi.org/project/atelier/
http://atelier.lino-framework.org/changes.html#version-1-1-9-released-2018-09-19

So you just need to say::

  pip install -U atelier

Sorry for the inconveniences!
Luc

**Comment by aralmpa** [comment]
Great! Everything works OK now!

The build succeeded. I will start writing right away!

Thank you Luc.
